### PR TITLE
WASM libraries bump

### PR DESCRIPTION
**Pull request by NikVolf**
along with somewhat extended error handling

### CODE REVIEW

File: ethcore/wasm/src/lib.rs [hunk 1]
```diff
@@ -70,14 +82,14 @@ impl vm::Vm for WasmInterpreter {
 	fn exec(&mut self, params: ActionParams, ext: &mut vm::Ext) -> vm::Result<GasLeft> {
 		let (module, data) = parser::payload(&params, ext.schedule().wasm())?;
 
-		let loaded_module = wasmi::Module::from_parity_wasm_module(module).map_err(Error)?;
+		let loaded_module = wasmi::Module::from_parity_wasm_module(module).map_err(Error::Interpreter)?;
 
 		let instantiation_resolover = env::ImportResolver::with_limit(16);
```
Comment (andresilva) [inline ethcore/wasm/src/lib.rs]: `instantiation_resolver`

File: ethcore/wasm/src/parser.rs [hunk 2]
```diff
@@ -22,14 +22,18 @@ use parity_wasm::elements::{self, Deserialize};
...
-		vals.insert(rules::InstructionType::Load, wasm_costs.mem as u32);
-		vals.insert(rules::InstructionType::Store, wasm_costs.mem as u32);
-		vals.insert(rules::InstructionType::Div, wasm_costs.div as u32);
-		vals.insert(rules::InstructionType::Mul, wasm_costs.mul as u32);
-		vals
-	}).with_grow_cost(wasm_costs.grow_mem)
+	rules::Set::new(
+		wasm_costs.regular,
+		{
+			let mut vals = ::std::collections::HashMap::with_capacity(8);
```
Comment (andresilva) [inline ethcore/wasm/src/parser.rs]: Why increase the capacity?

Comment (NikVolf) [reply ethcore/wasm/src/parser.rs]: Float disabling rules include 4 entries

File: Cargo.lock [hunk 3]
```diff
@@ -3509,24 +3514,25 @@ dependencies = [
...
  "log 0.3.8 (registry+https://github.com/rust-lang/crates.io-index)",
- "parity-wasm 0.23.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "parity-wasm 0.27.0 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
 [[package]]
 name = "wasmi"
 version = "0.0.0"
-source = "git+https://github.com/pepyakin/wasmi#551c99273042deaad869c17798060e2212deacab"
+source = "git+https://github.com/pepyakin/wasmi#f1a3f06d5e49798e2aa28150e863af1c7910bfc2"
```
Comment (tomusdrw) [inline Cargo.lock]: This is now in paritytech, right?

**Pull request merged**